app/Model/Server.php

```diff
@@ -4351,6 +4351,57 @@ public function checkoutMain()
         return exec('git checkout ' . $mainBranch);
     }
 
+    public function getSubmodulesGitStatus()
+    {
+        $submodulesNames = array('misp-galaxy', 'misp-taxonomies', 'misp-objects', 'misp-noticelist', 'misp-warninglists');
```

Comment:
What about PyMISP?

Reply:
Also, as you already have an acceptable submodule list, I would suggest to do the following for module names: 

```bash
git submodule |cut -f 3 -d\ 
```

This will return all the current submodules. (Which you could then also catch if none found etc...)